Lib/functools.py

```diff
@@ -856,9 +856,19 @@ def register(cls, func=None):
                 )
             func = cls
 
+            # bpo-41987 seems to only affect 3.9+
+            from sys import version_info
+
+            if version_info >= (3, 9) and isinstance(ann.get("return", None), str):
```

Comment:
Since this is the stdlib, you shouldn’t need to check the version.

Reply:
Understood :+1:  I've removed it

Reply:
Thanks. Are you planning to look into the root cause? (If you don't think you can I may be able to find time to look into this myself, but I can't promise anything.)